### PR TITLE
Cni network plugin for frakti

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Cni network plugin for frakti

### CODE REVIEW

**Comment by resouer** [comment]
Seems vendor dose not updated properly? @YaoZengzeng

File: pkg/hyper/hyper.go [hunk 1]
```diff
@@ -60,7 +61,12 @@ func NewHyperRuntime(hyperEndpoint string, streamingConfig *streaming.Config) (*
 		}
 	}
 
-	return &Runtime{client: hyperClient, streamingServer: streamingServer}, streamingServer, nil
+	networkPlugin, err := InitCNI()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &Runtime{client: hyperClient, streamingServer: streamingServer, networkPlugin: networkPlugin}, streamingServer, nil
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: nit: lines too long, consider format codes

File: pkg/hyper/network.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+	"github.com/containernetworking/cni/libcni"
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	DefaultInterfaceName = "eth0"
+	DefaultNetDir        = "/etc/cni/net.d"
+	DefaultCNIDir        = "/opt/cni/bin"
```
Comment (feiskyer) [inline pkg/hyper/network.go]: Could we make these two path configurable?

File: pkg/hyper/network.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+type CNINetworkPlugin struct {
+	sandboxNs map[string]*sandboxNetNS
+}
+
+func InitCNI() (*CNINetworkPlugin, error) {
+	plugin := &CNINetworkPlugin{}
+
+	plugin.sandboxNs = make(map[string]*sandboxNetNS)
```
Comment (feiskyer) [inline pkg/hyper/network.go]: Read and check network config here? So we could fail fast and don't need to read and check again and again later.

File: pkg/hyper/network.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+	}
+
+	for _, confFile := range files {
+		netconf, err := libcni.ConfFromFile(confFile)
+		if err != nil {
+			continue
+		}
+
+		rt := &libcni.RuntimeConf{
+			ContainerID: "cni",
```
Comment (feiskyer) [inline pkg/hyper/network.go]: Use sandboxID here?

File: pkg/hyper/network.go [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+func collectionInterfaceInfo() []*InterfaceInfo {
+	infos := []*InterfaceInfo{}
+
+	links, err := netlink.LinkList()
+	if err != nil {
+		return infos
+	}
+
+	for _, link := range links {
+		if link.Type() != "veth" {
```
Comment (feiskyer) [inline pkg/hyper/network.go]: In order to support other interface types, it's better to omit only `lo` here, e.g. `interface name == lo`

File: pkg/hyper/network.go [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+		return err
+	}
+
+	delete(plugin.sandboxNs, podname)
+
+	return nil
+}
+
+type sandboxNetNS struct {
+	netns  ns.NetNS
```
Comment (feiskyer) [inline pkg/hyper/network.go]: We need to persist sandbox netns, e.g. in pod labels or write to local files.

File: pkg/hyper/sandbox.go [hunk 7]
```diff
@@ -118,6 +156,11 @@ func (h *Runtime) DeletePodSandbox(podSandboxID string) error {
 		return err
 	}
 
+	err = h.networkPlugin.TeardownPodNetwork(podSandboxID)
```
Comment (feiskyer) [inline pkg/hyper/sandbox.go]: network should also be cleared up in `StopPodSandbox`

**Comment by YaoZengzeng** [comment]
This PR's code quality is not good enough, so close it and open another one #62 @resouer @feiskyer

**Pull request closed without merging**